ale_linters/haskell/hls.vim

```diff
@@ -0,0 +1,56 @@
+" Author: Yen3 <yen3rc@gmail.com>
+" Description: A language server for haskell
+"              The file is based on hie.vim (author: Luxed
+"              <devildead13@gmail.com>).  It search more project root files.
+
+call ale#Set('haskell_hls_executable', 'haskell-language-server-wrapper')
+
+function! ale_linters#haskell#hls#FindRootFile(buffer) abort
+    for l:path in ale#path#Upwards(expand('#' . a:buffer . ':p:h'))
+        if filereadable(l:path . 'stack.yaml')
+        \|| filereadable(l:path . 'cabal.project')
```

Comment:
it'd be nice to be able to choose the order in which files get searched (or to disable some altogether).

Reply:
...maybe I'm misunderstanding how this work and this just returns the directory the file is in

Reply:
I rewrite this part with a nested for loop. Hope it would be more clear.